Update calico.yaml for 1.3 branch

mogren opened the pull request:
*Description of changes:*

Based on #232, but includes a version bump to v3.3.6 to get the
[libcalico-go #1051](https://github.com/projectcalico/libcalico-go/pull/1051) fix.

For details, see https://docs.projectcalico.org/v3.3/releases/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

mogren commented:
@2ffs2nns I'm closing #232 in favor of this PR. I would really appreciate some help on how to verify this update works as intended.

2ffs2nns commented:
looking at it tonight, and running it through our CI. will post result when it's done. Thanks for picking it up

2ffs2nns commented on config/v1.3/calico.yaml:
```diff
@@ -35,7 +37,7 @@ spec:
         # host.
         - name: calico-node
-          image: quay.io/calico/node:v3.1.3
+          image: quay.io/calico/node:v3.3.5
```
should be `quay.io/calico/node:v3.3.6` to be current on v3.3.x

mogren replied:
Bumped it, thanks.

2ffs2nns commented on config/v1.3/calico.yaml:
```diff
@@ -363,67 +369,85 @@ spec:
-        name: calico-typha
-        ports:
-        - containerPort: 5473
+        - image: quay.io/calico/typha:v3.3.5
```
should be `quay.io/calico/typha:v3.3.6` to be current on v3.3.x

mogren replied:
Done

2ffs2nns commented:
@mogren manifest passed our CI w/3.3.6 images but after a manual diff of changes between v3.1.3 and v3.3.6, I want to suggest some more changes and run another test.

mogren commented:
@2ffs2nns Sure! I'm fine with that, thanks for testing.

2ffs2nns commented:
@mogren my git foo is failing me, there are multiple edits that need to be made to get this manifest in-sync with calico-v3.3.6 I've created a gist based off of your edits and posted here you can look at
https://gist.githubusercontent.com/2ffs2nns/53eed486b3040edefe0b1ed5295e8935/raw/b322cf3ff7d0f65fef6a76776b16ece17e374baa/et-eks

the edits I made
```
Add to lines 88-89
         resources:
            requests:
              cpu: 250m

Add to line 112-113
            - mountPath: /var/lib/calico
              name: var-lib-calico
              readOnly: false

Add to line 120
        - name: var-lib-calico
          hostPath:
            path: /var/lib/calico

Add to line 127
      tolerations:
        # Make sure calico-node gets scheduled on all nodes.
        - effect: NoSchedule
        - operator: Exists
        # Mark the pod as a critical add-on for rescheduling.
        - key: CriticalAddonsOnly
          operator: Exists
        - effect: NoExecute
          operator: Exists

Add to lines 376-377
      tolerations:
        # Mark the pod as a critical add-on for rescheduling.
        - key: CriticalAddonsOnly

Remove lines 409-412
          volumeMounts:
	            - mountPath: /etc/calico
	              name: etc-calico
	              readOnly: true

Remove lines 428-431
      volumes:
	        - name: etc-calico
	          hostPath:
	            path: /etc/calico
```

validation I completed w/above edits
* installed original calico-v3.1.3 based version w/EKS guestbook demo deployed
* performed rolling-upgrade from v3.1.3 w/new v3.3.6 based version (above gist)
* ran k8s NetworkPolicy e2e's

all looked good/passing. If you make these changes, I'll run through our CI again if you like w/your final changes

mogren commented on config/v1.4/calico.yaml:
```diff
@@ -177,7 +198,10 @@ metadata:
   scope: Cluster
   group: crd.projectcalico.org
-  version: v1
+  versions:
```
Using `version:` is deprecated, so I changed this according to the documentation https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition

mogren commented:
Thanks again @2ffs2nns, I added your changes and updated the config for 1.4 as well. Preparing to get a release out soon. The new changes are in a separate commit: https://github.com/aws/amazon-vpc-cni-k8s/pull/368/commits/ee13f3ef2b157e60bc01ab4692b76a22048022e7

2ffs2nns commented:
I ran the 1.3 file in `ee13f3e` through our CI. all good from my perspective
